app/tools.py

```diff
@@ -0,0 +1,88 @@
+import torch
+import torch.nn as nn
+import fortepyan as ff
+from tqdm import tqdm
+from omegaconf import OmegaConf, DictConfig
+
+from model import make_model
+from data.quantizer import MidiQuantizer
+from utils import vocab_sizes, greedy_decode
+from data.dataset import quantized_piece_to_records
+from data.tokenizer import VelocityEncoder, QuantizedMidiEncoder
+
+
+def encode_record(sequence: dict, encoder: QuantizedMidiEncoder) -> torch.Tensor:
+    token_ids = encoder.encode(sequence)
+    token_ids = [encoder.token_to_id["<CLS>"]] + token_ids
+    return torch.tensor(token_ids, dtype=torch.int64)
+
+
+def process_piece(train_cfg: DictConfig, piece: ff.MidiPiece, quantizer: MidiQuantizer) -> list[torch.Tensor]:
+    """
+    Run full pre-processing on a piece
+    """
+    src_encoder = QuantizedMidiEncoder(train_cfg.dataset.quantization)
+
+    # pre-process the piece ...
+    qpiece = quantizer.inject_quantization_features(piece)
+    sequences = quantized_piece_to_records(
+        piece=qpiece,
+        sequence_len=train_cfg.dataset.sequence_len,
+        sequence_step=train_cfg.dataset.sequence_len,
+    )
+    sequence_tokens = [encode_record(sequence, src_encoder) for sequence in sequences]
```

Comment:
```suggestion
    sequence_tokens = [src_encoder.encode_record(sequence) for sequence in sequences]
```
see the comment about `encode_record` :arrow_up: